vars/tdr.groovy

```diff
@@ -0,0 +1,11 @@
+def buildEndToEndTests(int delay, String buildJob, String[] params) {
```

Comment:
Can `delay` be `delaySeconds` (if that's what the units are)?

It's a good principle to follow that if you ever have a number representing a measurement like time, it should have units in the name to stop people from passing in a number that's completely the wrong order of magnitude. 🙂 

(Ideally, you'd make this completely unambiguous by using the type system - you're less likely to get the units wrong if you use a `Duration` rather than an `int`, but that's not going to be easy in a Groovy Jenkinsfile so let's make it clear with the parameter name.)

Reply:
> I've added quite a few comments, but don't panic.  They're mostly suggestions that I think will be helpful to people calling these functions.
> 
> The general principle behind the suggestions is that we're wrapping lower-level functionality with higher-level helper functions, so we should make it as obvious as possible how to use those functions without having to look up documentation. You're already doing this, e.g. by calling the parameter `delay` rather than `quietPeriod`.  I've just suggested a few more.

Thanks for having a look! I'll work on the changes you suggestsed. I'm also working on a README for this repo that has some more details about the functions/their uses etc to give some more context :ok_hand:

Reply:
Might be worth considering if the docs you're writing would fit better as doc comments on the functions, rather than in the readme? In a library of ad-hoc helper functions like this, people are quite likely to skim the code rather than looking in the readme.